### PR TITLE
Fr datasource

### PR DESCRIPTION
**Pull request by AnikaAgiwal2711**
# Description
A few sentences describing the overall goals of the pull request's commits.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 80% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

### CODE REVIEW

File: docs/data-sources/firmware_repository.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
+---
+# Copyright (c) 2023-2024 Dell Inc., or its subsidiaries. All Rights Reserved.
```
Comment (shenda1) [inline docs/data-sources/firmware_repository.md:2]: 2024

File: powerflex/helper/firmware_repository_helper.go [hunk 2]
```diff
@@ -35,3 +35,128 @@ func UpdateFrimwareRepositoryState(frDetails *scaleiotypes.UploadComplianceTopol
...
+		Custom:              types.BoolValue(input.Custom),
+		NeedsAttention:      types.BoolValue(input.NeedsAttention),
+		Ignore:              types.BoolValue(input.Ignore),
+		OriginalComponentID: types.StringValue(input.OriginalComponentID),
+		FirmwareRepoName:    types.StringValue(input.FirmwareRepoName),
+		SystemIDs:           newSystemIDS(input.SystemIDs),
+	}
+}
+
+// // newtypes.StringList converts list of client.types.String to list of models.types.String
```
Comment (shenda1) [inline powerflex/helper/firmware_repository_helper.go:123]: remove if not required

File: powerflex/provider/firmware_repository_datasource_schema.go [hunk 3]
```diff
@@ -0,0 +1,453 @@
+/*
+Copyright (c) 2023-2024 Dell Inc., or its subsidiaries. All Rights Reserved.
```
Comment (shenda1) [inline powerflex/provider/firmware_repository_datasource_schema.go:2]: 2024

File: powerflex/provider/firmware_repository_datasource_schema.go [hunk 4]
```diff
@@ -0,0 +1,453 @@
...
+import (
+	"github.com/hashicorp/terraform-plugin-framework-validators/setvalidator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+// FaultSetDataSourceSchema defines the schema for Fault Set datasource
```
Comment (shenda1) [inline powerflex/provider/firmware_repository_datasource_schema.go:29]: FaultSetDataSourceSchema ->  FirmwareRepositoryDataSourceSchema

File: powerflex/provider/firmware_repository_datasource_test.go [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+import (
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+var FRDataSourceConfig1 = `
+data "powerflex_firmware_repository" "test" {
+	firmware_repository_ids = ["8aaa3fda8f5c2609018f854266e12865", "8aaa3fda8f5c2609018f857b6c0d2ede"]
```
Comment (shenda1) [inline powerflex/provider/firmware_repository_datasource_test.go:29]: Don't hardcode the IDs or Names

File: powerflex/provider/firmware_repository_datasource_test.go [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+}
+`
+
+func TestAccFRDataSource(t *testing.T) {
+	t.Skip("Skipping this test case")
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: ProviderConfigForTesting + FRDataSourceConfig1,
```
Comment (shenda1) [inline powerflex/provider/firmware_repository_datasource_test.go:62]: Please add assertion statements

**Review by shenda1 — CHANGES_REQUESTED** [review]
Add the coverage report

**Comment by AnikaAgiwal2711** [comment]
image

**Comment by AnikaAgiwal2711** [comment]
image (6)

File: docs/data-sources/firmware_repository.md (hunk 1 above)
Comment (AnikaAgiwal2711) [reply docs/data-sources/firmware_repository.md]: Done

File: powerflex/helper/firmware_repository_helper.go (hunk 2 above)
Comment (AnikaAgiwal2711) [reply powerflex/helper/firmware_repository_helper.go]: Done

File: powerflex/provider/firmware_repository_datasource_schema.go (hunk 3 above)
Comment (AnikaAgiwal2711) [reply powerflex/provider/firmware_repository_datasource_schema.go]: Done

File: powerflex/provider/firmware_repository_datasource_schema.go (hunk 4 above)
Comment (AnikaAgiwal2711) [reply powerflex/provider/firmware_repository_datasource_schema.go]: Done

File: powerflex/provider/firmware_repository_datasource_test.go (hunk 5 above)
Comment (AnikaAgiwal2711) [reply powerflex/provider/firmware_repository_datasource_test.go]: Done

**Comment by AnikaAgiwal2711** [comment]
run e2e test

**Comment by Krunal-Thakkar** [comment]
run e2e test

**Comment by Krunal-Thakkar** [comment]
run e2e test

**Comment by shenda1** [comment]
run e2e test

**Comment by Krunal-Thakkar** [comment]
run e2e test

File: powerflex/provider/firmware_repository_resource_test.go [hunk 7]
```diff
@@ -0,0 +1,158 @@
+/*
+Copyright (c) 2023-2024 Dell Inc., or its subsidiaries. All Rights Reserved.
```
[the remark this answers is not in the document]
Comment (AnikaAgiwal2711) [reply powerflex/provider/firmware_repository_resource_test.go]: Done

File: powerflex/provider/firmware_repository_datasource_test.go [hunk 8]
```diff
@@ -0,0 +1,98 @@
+/*
+Copyright (c) 2023-2024 Dell Inc., or its subsidiaries. All Rights Reserved.
```
[the remark this answers is not in the document]
Comment (AnikaAgiwal2711) [reply powerflex/provider/firmware_repository_datasource_test.go]: Done

**Comment by AnikaAgiwal2711** [comment]
run e2e test

**Review by shenda1 — APPROVED** [review]

**Review by Krunal-Thakkar — APPROVED** [review]

**Pull request merged**